### PR TITLE
App insights

### PR DESCRIPTION
**Pull request by doncem**
JIRA:
- [Spike: Centralised Logging in Azure](https://tools.hmcts.net/jira/browse/RPE-232)
- [Metrics](https://tools.hmcts.net/jira/browse/RPE-210)

Module includes all the Azure App Insights default glories:
- request filter/interceptor
- built in performance counters
- web modules
- core and web intialisers
- agent jar publication (microsoft does not provide as artifact anywhere - need to download manually)

### CODE REVIEW

File: java-logging-appinsights/README.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+- TimestampPropertyInitializer
+- WebOperationIdTelemetryInitializer
+- WebOperationNameTelemetryInitializer
+- WebSessionTelemetryInitializer
+- WebUserTelemetryInitializer
+- WebUserAgentTelemetryInitializer
+
+#### Debug mode
+
+By default debug mode is on. To turn it off set in configuration manually (for request filter and interceptor to pick it up):
```
Comment (timja-kainos) [inline java-logging-appinsights/README.md]: what does the debug mode do and does it need to be on by default?

Comment (doncem) [reply java-logging-appinsights/README.md]: why did i call debug mode? it's developer mode. [explanation](https://github.com/MicrosoftDocs/azure-docs/blob/master/articles/application-insights/app-insights-configuration-with-applicationinsights-config.md#developer-mode)

Comment (timja-kainos) [reply java-logging-appinsights/README.md]: >  It causes significant overhead in CPU and network bandwidth.
Could we turn it off by default?

Comment (doncem) [reply java-logging-appinsights/README.md]: Can remove entirely

File: java-logging-appinsights/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+By default developer mode is off. To turn it on set in configuration manually (for telemetry client to pick it up):
+
+```java
+TelemetryConfiguration.getActive().getChannel().setDeveloperMode(true);
+```
+
+or simply turn on by configuration:
+
+```yaml
```
Comment (timja-kainos) [inline java-logging-appinsights/README.md]: this isn't yaml syntax is it?

Comment (doncem) [reply java-logging-appinsights/README.md]: nope, property. i'll change to yaml

**Pull request merged**